### PR TITLE
Refactor mock_system class for clarity

### PR DESCRIPTION
**Pull request by mszulcz-mitre**
The mock_system class has a few parts that could cause confusion. For example, the words 'module' and 'component' are used interchangeably, it's not clear when modules will be mocked, and it's not clear what the return value of start_servers means (see Issue #183).  This commit refactors the code to clarify these issues.

Signed-off-by: Michael L. Szulczewski <mszulczewski@mitre.org>

### CODE REVIEW

File: tests/integration/mock_system.cpp [hunk 1]
```diff
@@ -31,8 +31,44 @@ namespace cbdc::test {
...
+        for(auto& module_and_endpoints : m_module_endpoints) {
+            const auto& endpoints = module_and_endpoints.second;
+            if(endpoints.empty()) {
+                const auto& module = module_and_endpoints.first;
+                m_logger->warn(
+                    "The " + mock_system_module_string(module)
+                    + " module was selected to be mocked,\n"
+                      "but can't be because no endpoint was given.");
+                m_module_endpoints.erase(module);
+            }
```
Comment (maurermi) [inline tests/integration/mock_system.cpp:71]: Adds clarity for which modules are being tested

File: tests/integration/mock_system.hpp [hunk 2]
```diff
@@ -55,35 +55,35 @@ namespace cbdc::test {
...
         template<typename T>
         auto expect(mock_system_module for_module,
                     std::optional<cbdc::buffer>&& reply_with = std::nullopt,
-                    uint64_t component_id = 0) -> std::future<T> {
+                    uint64_t module_id = 0) -> std::future<T> {
             auto res_promise = std::make_shared<std::promise<T>>();
             auto res_future = res_promise->get_future();
             const std::lock_guard<std::mutex> guard(m_handler_lock);
-            auto idx = std::make_pair(for_module, component_id);
+            auto idx = std::make_pair(for_module, module_id);
```
Comment (maurermi) [inline tests/integration/mock_system.hpp:86]: This rename is appropriate, and improves clarity.

File: tests/integration/mock_system.hpp [hunk 3]
```diff
@@ -129,13 +129,9 @@ namespace cbdc::test {
...
-        std::unordered_set<mock_system_module> m_modules{
-            mock_system_module::watchtower,
-            mock_system_module::atomizer,
-            mock_system_module::coordinator,
-            mock_system_module::archiver,
-            mock_system_module::shard,
-            mock_system_module::sentinel};
+        std::unordered_map<mock_system_module,
+                           std::vector<cbdc::network::endpoint_t>>
+            m_module_endpoints;
```
Comment (maurermi) [inline tests/integration/mock_system.hpp:134]: Map is clearer to understand in this context

**Review by maurermi — COMMENTED** [review]
T-ACK. This PR refactors mock_system such to improve usage clarity and documentation. No significant effect on current performance or usage, but should make future tests easier to implement.

**Review by HalosGhost — APPROVED** [review]
tACK. Pleasant clarity improvement for the `mock_system`. Merging.

**Pull request merged**